src/pages/AppDetails/AppDetailsPage.tsx

```diff
@@ -0,0 +1,164 @@
+import * as React from 'react';
+import * as API from '../../services/Api';
+import { Link, RouteComponentProps } from 'react-router-dom';
+import { AppId, App } from '../../types/App';
+import { authentication } from '../../utils/Authentication';
+import { Breadcrumb, TabContainer, Nav, NavItem, TabContent, TabPane } from 'patternfly-react';
+import AppInfo from './AppInfo';
+import * as MessageCenter from '../../utils/MessageCenter';
+import AppMetricsContainer from '../../containers/AppMetricsContainer';
+import { AppHealth } from '../../types/Health';
+
+type AppDetailsState = {
+  app: App;
+  health?: AppHealth;
+};
+interface ParsedSearch {
+  type?: string;
+  name?: string;
+}
+
+class AppDetails extends React.Component<RouteComponentProps<AppId>, AppDetailsState> {
+  constructor(props: RouteComponentProps<AppId>) {
+    super(props);
+    this.state = {
+      app: {
+        namespace: { name: '' },
+        name: '',
+        workloads: []
+      }
+    };
+    this.fetchApp();
+  }
+
+  appPageURL(parsedSearch?: ParsedSearch) {
+    return '/namespaces/' + this.props.match.params.namespace + '/applications/' + this.props.match.params.app;
+  }
+
+  fetchApp = () => {
+    let promiseDetails = API.getApp(authentication(), this.props.match.params.namespace, this.props.match.params.app);
+
+    let promiseHealth = API.getAppHealth(
+      authentication(),
+      this.props.match.params.namespace,
+      this.props.match.params.app,
+      600
+    );
+
+    Promise.all([promiseDetails, promiseHealth])
+      .then(([resultDetails, resultHealth]) => {
+        this.setState({
+          app: resultDetails.data,
+          health: resultHealth
+        });
+      })
+      .catch(error => {
+        MessageCenter.add(API.getErrorMsg('Could not fetch App Details.', error));
+      });
+  };
+
+  renderBreadcrumbs = () => {
+    const urlParams = new URLSearchParams(this.props.location.search);
+    const to = this.appPageURL();
+    let tab = 'Info';
+    switch (urlParams.get('tab')) {
+      case 'info':
+        tab = 'Info';
+        break;
+      case 'in_metrics':
+        tab = 'Inbound Metrics';
+        break;
+      case 'out_metrics':
+        tab = 'Outbound Metrics';
+        break;
+      default:
+        tab = 'Info';
+    }
+    return (
+      <Breadcrumb title={true}>
+        <Breadcrumb.Item componentClass="span">
+          <Link to="/applications">Applications</Link>
```

Comment:
you might be interested on clear filters `onClick` otherwise you will be filtering apps by filters stored in `NamespaceFilterSelected`